### ISSUE TITLE
search: url with sortby param doesn't return sorted results

### ISSUE THREAD

**Issue by ascott** [participedia/api#566]
/?sortby=post_date and /?sortby=updated_at should return sorted results but does not.

**Comment by dethe** [comment]
Whichever way we sort, featured articles should be first, right?

**Comment by jesicarson** [comment]
Yes.

**Comment by jesicarson** [comment]
@dethe can we make the card display say "Edited" before the date if you sort by last edited?

**Comment by dethe** [comment]
Sort order only makes sense for when there is no search query, which is the the default search, for featured articles. If there is a search query we sort by relevance.

@jesicarson We can change the results card to show the edited timestamp and label instead of created. @ascott what would be your preferred way to handle that? Different partial, or one partial with a flag? Alternatively, we can show both timestamps on the card.

**Comment by jesicarson** [comment]
Would using the sort by drop down clear your current search query if you had one? What if you wanted to sort the results from your current search by edited date? @scottofletcher do you care? this is your baby. 

Regarding the display - yes it would be good to change the label to "edited" if sorting by edited date. It should still say "submitted" when the results are displayed in order of the date they are submitted - I think this is the default on the homepage and tabs if no search query is entered.

**Comment by ascott** [comment]
@dethe one partial with a flag would be my preference

> If there is a search query we sort by relevance.
in this case wouldn't we sort by relevance and then by the sortBy filter? but if we can't sort on results with a query we should hide the sort by drop down when there is a query term. it feels like this could be a confusing ui though. most users won't realize why they can sort by filters sometimes and not other times.

**Comment by scottofletcher** [comment]
on the first point about featured entries: performing any search/sort should get rid of them. featured entries should only come first on the homepage (unless, of course, they are the most relevant entry to whatever the user searched for). 

on the issue of when to allow sort by: I agree with @dethe that sort by doesn't make sense when you're using the search bar. when you're doing a keyword search, the most relevant entries should be returned first. however, even though it doesn't make sense, I still think that we should allow people to do it. think of this situation: someone doesn't know how the faceted search works, so they use a keyword search to look for cases that use the 'town hall meetings' method. because they're using the keyword search like a faceted search, they might then use the sort by function. in which case I agree with @ascott - it's going to make the ui confusing if we either hide the sort by filter or we clear their results as soon as they use it. 

TL;DR: 1) featured content only on homepage, and 2) always allow people to access the sort by function.

regarding labels: I agree with @jesicarson: if you've sorted by last edited (or updated or whatever term we're using), the cards should say 'Edited' instead of 'Submitted' with the date of last changes

**Comment by jesicarson** [comment]
@dethe 

1. if you have already started building sort by with featured first in results, you can leave it as is. if it is easy to change, and you agree it makes sense to change it to not show featured first, please go ahead.

2. is it possible to sort an existing set of results by date submitted/edited instead of by relevance? if yes, and it makes sense to implement (ie: won't take extra time), please go ahead. if you have a different solution that is the path of least resistance to get the sort by functioning, let us know.

**Comment by dethe** [comment]
All search results are "on the homepage" in that they use that layout. Featured content searches are the default search, so that's what shows up when they visit the homepage and have not searched for anything. Making a full text search clears the default search by featured.

Search results get sorted by relevance, which is something like a number between zero and one, i.e., 0.812 is more relevant than 0.424. Sorting by relevance, then date probably won't have much in the way of visible effect, unless a lot of search results end up with the exact same relevance, which seems unlikely.

If we're going to let people sort by date on search results, we should include relevance in the sortby dropdown, so they can get back to it. Of course, that won't make any sense for featured results.

For what it's worth, featured is a faceted search, so all of these issues are probably going to be even more relevant when we have more ways of performing faceted searches. Will we ever allow faceted searches to be combined with full-text searches? Because currently they are mutually exclusive (it is either full-text search OR featured search), but I could see wanting them to work together in the future.

An alternative to hiding searches that don't make sense, would be to disable them and grey them out. The UI doesn't change around in unpredictable ways, but it clearly shows you can't do that right now. We can even have a tool tip that says why.

**Comment by dethe** [comment]
@jesicarson 
1. Featured search has always sorted by featured first, then by date. It uses the created date by default. I've fixed the backend so it can sort by last edited date instead, but keeps the featured flag as the primary sort. If sorting by date is *instead* of by featured, rather than in addition to it, then we should add "featured" to the drop-down so users can get back to it.

2. It is possible to sort search results by date even if it doesn't make sense. That is not how it works right now, so it will take some time to implement. Probably not a lot of time, but a couple of hours to write and test.

**Comment by scottofletcher** [comment]
RE sort by on search results: if it's easy to grey out the function instead of hiding it, then I'd go with that option since it sounds like it gets really complicated otherwise. 

RE sort by vs featured: I don't think users will understand what 'featured' means in the drop-down, so I guess we should just always keep featured content at the top (I'll have to make sure I rotate the entries enough so people don't have to look at the same ones for an extended period of time!)

RE faceted + full text search: ideally they would be combined. that way people can include terms that aren't included in the faceted search options (eg. we leave out specific topics, so I could see someone selecting 'Environment' from General Topics and then doing a keyword search for 'climate change'

**Comment by dethe** [comment]
OK, this sounds good to me, and not just because it is how the back end works already.

Part of the reason full text search and featured search (the only facet of faceted search we have currently) are mutually exclusive is because we don't actually expose featured search in the UI aside from marking the featured articles as featured. So it is a hidden, UI-less search, which the other facets won't be. Whether sorting options make sense for full-text + faceted search is another question we'll have to sort out. Faceted searches are really filters, so if there are filters by themselves, then perhaps choosing a sorting field makes sense, but filters over full-text search probably do not need additional sorting.

There is also the question of whether all results are returned (as for featured) or only the results which match the search + filtering criteria, but that conversation should probably be had over on th #586 issue.

**Comment by scottofletcher** [comment]
RE additional sorting: if it makes more sense to disable the sort by feature whenever someone does a full text search (alone or in combination w/faceted), then maybe go with that

RE returning results: I would say only the results which match the search + filtering criteria (treat them like "search term 1" AND "search term 2" AND (etc.)

**Comment by jesicarson** [comment]
Sorry I have to sign off and cant review this today. dethe, do you need answers now or can we chat about this on wednesdays call? sort by is not a critical issue for launch, and it can be moved down on the priorities list.
